src/data/actions/assets.js

```diff
@@ -121,46 +121,44 @@ export const uploadingAssets = count => ({
   count,
 });
 
-export const uploadAssetSuccess = (asset, response) => ({
+export const uploadAssetSuccess = response => ({
   type: assetActions.UPLOAD_ASSET_SUCCESS,
-  asset,
   response,
 });
 
-export const uploadAssetFailure = (file, response) => ({
+export const uploadAssetFailure = (asset, response) => ({
   type: assetActions.UPLOAD_ASSET_FAILURE,
-  file,
+  asset,
   response,
 });
 
 export const uploadAssets = (files, courseDetails) =>
   (dispatch, getState) => {
     dispatch(uploadingAssets(files.length));
-    files.forEach((file) => {
+    // gather all the promises into a single promise that can be returned
+    return Promise.all(files.map(file => (
       clientApi.postUploadAsset(courseDetails.id, file)
         .then((response) => {
           if (response.ok) {
-            response.json().then((data) => {
-              dispatch(uploadAssetSuccess(data.asset, response));
-              dispatch(getAssets(getState().request, courseDetails));
+            return response.json().then((json) => {
+              dispatch(uploadAssetSuccess(json));
```

Comment:
I've changed the uploadAssetSuccess action creator to just take the JSONified response, since the raw Response isn't super useful, and asset happens to be a part of the response JSON anyway.